File: app/src/main/java/com/example/networkaplication/home/search/story/SearchStoryPresenter.java
package com.example.networkaplication.home.search.story;

import java.util.ArrayList;

public class SearchStoryPresenter {

    private ArrayList<SearchItem> items;

    public SearchStoryPresenter(ArrayList<SearchItem> items) {
        this.items = items;
    }

    public void onBindSearchStoryRowAtPosition(SearchStoryRowView rowView, int position) {
        SearchItem item = items.get(position);
        rowView.setSearchStoryTest(item.getSearchText());
    }

    public int getSearchRowCount() {
        return items.size();
    }

    public SearchItem getSearchItem(int position) {
        return items.get(position);
    }

    public void refreshSearch(ArrayList<SearchItem> refreshList) {
        this.items = new ArrayList<>(refreshList);
    }
}

File: app/src/main/java/com/example/networkaplication/home/search/story/SearchStoryPresenter.kt
package com.example.networkaplication.home.search.story

import java.util.ArrayList

class SearchStoryPresenter(private var items: ArrayList<SearchItem>?) {

    internal val searchRowCount: Int
        get() = items!!.size

    internal fun onBindSearchStoryRowAtPosition(rowView: SearchStoryRowView, position: Int) {
        val item = items!![position]
        rowView.setSearchStoryTest(item.searchText!!)
    }

    internal fun getSearchItem(position: Int): SearchItem {
        return items!![position]
    }

    internal fun refreshSearch(refreshList: ArrayList<SearchItem>) {
        this.items = ArrayList(refreshList)
    }
}
